app/src/main/kotlin/com/wire/android/navigation/NavigationItem.kt

```diff
@@ -168,15 +169,20 @@ enum class NavigationItem(
 
     OtherUserProfile(
         primaryRoute = OTHER_USER_PROFILE,
-        canonicalRoute = "$OTHER_USER_PROFILE/{$EXTRA_USER_DOMAIN}/{$EXTRA_USER_ID}",
+        canonicalRoute = "$OTHER_USER_PROFILE?$EXTRA_USER_ID={$EXTRA_USER_ID}",
+        deepLinks = listOf(navDeepLink {
+            uriPattern = "${DeepLinkProcessor.DEEP_LINK_SCHEME}://" +
+                    "${DeepLinkProcessor.OTHER_USER_PROFILE_DEEPLINK_HOST}/" +
+                    "{$EXTRA_USER_ID}"
+        }),
         content = { OtherUserProfileScreen() },
         animationConfig = NavigationAnimationConfig.NoAnimation
     ) {
         override fun getRouteWithArgs(arguments: List<Any>): String {
-            val userDomain: String = arguments.filterIsInstance<String>()[0]
-            val userProfileId: String = arguments.filterIsInstance<String>()[1]
-
-            return "$primaryRoute/$userDomain/$userProfileId"
+            val userIdString: String = checkNotNull(
+                arguments.filterIsInstance<QualifiedID>().firstOrNull()?.toString()
+            ) { "Required UserId is null." }
```

Comment:
This will still cause a crash on the app with that exception, is that what we want?

Reply:
no, i reverted it :)